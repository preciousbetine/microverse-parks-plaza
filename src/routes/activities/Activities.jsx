import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { LinkContainer } from 'react-router-bootstrap';
import Styles from '../../assets/styles/activities.module.scss';
import { fetchActivities } from '../../redux/slices/activitySlice';
import ThemeSwitch from '../../components/ThemeSwitch';

function Activities() {
  const loading = useSelector((store) => store.activities.loading);
  const activities = useSelector((store) => store.activities.activities);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!loading && !activities.length) dispatch(fetchActivities());
  });

  return (
    <div className={Styles['activities-page']}>
      <header className={Styles.header}>
        <div>
          <i className="fa-solid fa-square-parking" />
          <h1>PARKS PLAZA</h1>
        </div>
        <ThemeSwitch />
      </header>
      <main>
        <section className={Styles.title}>
          <div className={Styles.bg}>
            {
              'AMERICA PARKS'.repeat(10)
            }
          </div>
          <div className={Styles['header-text']}>
            <h2>USA</h2>
            <p>
              {activities.length}
              {' '}
              activities
            </p>
          </div>
        </section>
        <section className={Styles.activities}>
          <h3>
            <i className="fa-regular fa-rectangle-list" />
            &nbsp;
            ALL ACTIVITIES
          </h3>
          {
            loading && (
              <div className={Styles.loader} />
            )
          }
          <ul>
            {
              !loading && (
                activities.map((activity) => (
                  <LinkContainer key={activity.id} to={`/parks/${activity.id}`}>
                    <li>
                      <span className={Styles['activity-bg']}>{activity.name.toUpperCase()[0]}</span>
                      <i className="fa-regular fa-circle-right" />
                      <span className={Styles['activity-name']}>{activity.name.toUpperCase()}</span>
                    </li>
                  </LinkContainer>
                ))
              )
            }
          </ul>
        </section>
      </main>
    </div>
  );
}

export default Activities;
