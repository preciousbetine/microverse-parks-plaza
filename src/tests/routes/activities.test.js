/**
 * @jest-environment jsdom
 */

import React from 'react';
import { MemoryRouter } from 'react-router';
import Activities from '../../routes/activities/Activities';
import renderWithProviders from '../utils';

describe('Activities page', () => {
  it('renders correctly', () => {
    const { page } = renderWithProviders(
      <MemoryRouter>
        <Activities />
      </MemoryRouter>,
    );
    expect(page).toMatchSnapshot();
  });
});
