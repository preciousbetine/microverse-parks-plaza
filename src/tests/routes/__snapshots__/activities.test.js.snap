// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activities page renders correctly 1`] = `undefined`;
